### PR TITLE
Fix admin forgot password config with non default email providers

### PR DESCRIPTION
**Pull request by felixhamel**
#### Description of what you did:

- [x] Copy the code from **strapi-plugin-users-permissions/controllers/Auth.js** into **node_modules/strapi-admin/controllers/Auth.js** to retrieve email config from database instead of hard coded settings.

#### My PR is a:

- [ ] 💥 Breaking change
- [X] 🐛 Bug fix
- [ ] 💅 Enhancement
- [ ] 🚀 New feature

#### Main update on the:

- [X] Admin
- [ ] Documentation
- [ ] Framework
- [ ] Plugin

#### Manual testing done on the following databases:

- [ ] Not applicable
- [X] MongoDB
- [ ] MySQL
- [ ] Postgres
- [ ] SQLite

### CODE REVIEW

**Comment by lauriejim** [comment]
Hello @felixhamel ! You describe what you did, thank you.
Can you tell me why, for what this update is made?
Thank you.

**Comment by felixhamel** [comment]
Hi @lauriejim,

This update is made for users that use a custom email provider and use the admin login page to reset their password. The current code have an hardcoded settings that push a different content and (to/from) param than the one in the database. The API call works, but this one don't use the same settings than the API one, which cause problem here.

In my use case, we use AWS SES to send our email and the service don't like that we try to send an email with a "from" = "no-reply@strapi.io". In the database config, we use a custom to, from and content that should be used.

**Comment by lauriejim** [comment]
Hello @felixhamel Did you test your update?

**Comment by felixhamel** [comment]
> Hello @felixhamel Did you test your update?

Sure. I tested it with the SES plugin and I tried with the instructions in https://github.com/felixhamel/strapi/blob/master/CONTRIBUTING.md with sendmail.

**Comment by lauriejim** [comment]
Okay, but the URL you send is not correct and doesn't redirect to the right page.
That is the reason why I'm confused about your PR.

**Comment by felixhamel** [comment]
You are right that for administrators the URL to validate the account is not the right one. I'll check to fix it this week.

**Comment by lauriejim** [comment]
Hello! Any news about your updates?

**Comment by felixhamel** [comment]
> Hello! Any news about your updates?

Hi! Got a bit busy with the holidays and a newborn. I'll do it this weekend ;)

**Comment by felixhamel** [comment]
@lauriejim I have updated the code. We use the same email configuration as the one defined in the admin panel, but with a default email message to keep it separated from the "user" flow as the URL is not the same. This allow both the api reset password and the admin reset password to use the external email plugin as configured if wanted.

**Comment by lauriejim** [comment]
@Aurelsicoko and @alexandrebodin I have a question for you about this PR.

The **email template** that is used in this PR is the one defined in the Admin Panel.
But this email template is for the **User** experience (https://strapi.io/documentation/3.0.0-beta.x/plugins/users-permissions.html#forgotten-password) 
I think the message template between an **Administrator** that use the Admin Panel and a User should be different.

 What do you think about that.

**Comment by alexandrebodin** [comment]
I don't mind that mesage but we could make it more specific for Admins.

**Comment by alexandrebodin** [comment]
Hey @felixhamel can you update your PR ?

File: packages/strapi-admin/controllers/Auth.js [hunk 1]
```diff
@@ -321,21 +327,50 @@ module.exports = {
...
 
-    const settings = {
-      from: {
-        name: 'Administration Panel',
-        email: 'no-reply@strapi.io',
-      },
-      response_email: '',
-      object: 'Reset password',
-      message: `<p>We heard that you lost your password. Sorry about that!</p>
+    // Retrieve email settings from database. Enable the use of external email plugins.
```
Comment (lauriejim) [inline packages/strapi-admin/controllers/Auth.js]: Can you update this comment and remove the second sentence, please.
This not impact the usage of email plugin itself.

Comment (felixhamel) [reply packages/strapi-admin/controllers/Auth.js]: Done 😄

**Comment by lauriejim** [comment]
Can you please remind me of the exact "why" of your update.
I still not sure to fully understand.

**Comment by felixhamel** [comment]
This PR is only for the ForgotPassword call done by the UI login. Currently, it ignore the setting put in the administration for normal user email sending. So if I set the SES plugin, it won't use it with the right config. That way, there is a high chance of failing to send the email with weird errors (our case).

To fix it, I took the code from the ForgotPassword for non administrator where it pull email config from the DB and use it to send the email with the right configuration and the right plugin.

**Comment by alexandrebodin** [comment]
Hi, THis PR uses the `users-permissions` plugin but the code is in the Admin and should not depend on this plugin. We are going to refactor completely the Admin and will come up with a solution to configure the emailing for the forgot password email of the administrators. 

I'm going to close this PR in the meantime. Tell me if you have some other requirements for this new feature :) 

Thanks a lot for your efforts !

**Comment by giacomocerquone** [comment]
@alexandrebodin Hi, do you came up with a solution at the end?

**Comment by hasindur-digital400** [comment]
@alexandrebodin any solution to this problem? I'm using Strapi V4.20.1 and still facing the issue.

**Comment by nataliarosa9** [comment]
@alexandrebodin, any solution to this problem? I'm using Strapi v4.25.9 and am still facing the issue.

**Comment by alexandrebodin** [comment]
Hello, this issue is from 4 years ago please re-open a new issue with the relevant information so the team can reproduce it and triage it 🙏

**Pull request closed without merging**